examples/task_incremental_with_checkpointing.py

```diff
@@ -0,0 +1,235 @@
+################################################################################
+# Copyright (c) 2021 ContinualAI.                                              #
+# Copyrights licensed under the MIT License.                                   #
+# See the accompanying LICENSE file for terms.                                 #
+#                                                                              #
+# Date: 01-12-2020                                                             #
+# Author(s): Andrea Cossu                                                      #
+# E-mail: contact@continualai.org                                              #
+# Website: avalanche.continualai.org                                           #
+################################################################################
+"""
+Example on how to use the checkpoint plugin.
+
+This is basically a vanilla Avalanche main script, but with the replay
+functionality enabled. Proper comments are provided to point out the changes
+required to use the checkpoint plugin.
+"""
+
+import argparse
+from typing import Sequence
+
+import torch
+from torch.nn import CrossEntropyLoss
+from torch.optim import SGD
+
+from avalanche.benchmarks import CLExperience, SplitCIFAR100, CLStream51, \
+    SplitOmniglot
+from avalanche.benchmarks.classic import SplitCIFAR10
+from avalanche.evaluation.metrics import accuracy_metrics, loss_metrics, \
+    class_accuracy_metrics
+from avalanche.logging import InteractiveLogger, TensorboardLogger, \
+    WandBLogger
+from avalanche.models import SimpleMLP, as_multitask
+from avalanche.training.determinism.rng_manager import RNGManager
+from avalanche.training.plugins import EvaluationPlugin, CWRStarPlugin, \
+    ReplayPlugin, GDumbPlugin, LwFPlugin, SynapticIntelligencePlugin, EWCPlugin
+from avalanche.training.plugins.checkpoint import CheckpointPlugin, \
+    FileSystemCheckpointStorage
+from avalanche.training.supervised import Naive
+
+
+def main(args):
+    # FIRST CHANGE: SET THE RANDOM SEEDS
+    # In fact, you should to this no matter the checkpointing functionality.
+    # Using `use_deterministic_algorithms` may require setting the
+    # CUBLAS_WORKSPACE_CONFIG=:4096:8 environment variable.
+
+    # Remember to load checkpoints by setting the same random seed used when
+    # creating them...
+    RNGManager.set_random_seeds(1234)
+    torch.use_deterministic_algorithms(True)
```

Comment:
do we have to use deterministic algorithms? Because those are often slower. From pytorch docs: "Deterministic operations are often slower than nondeterministic operations, so single-run performance may decrease for your model. "

Reply:
They are not mandatory. Determinism is needed in the example to check if the checkpointing plugin can correctly resume the training. I'll add an explanation to the examples.